test/bats/test.bats

```diff
@@ -81,6 +86,17 @@ teardown() {
   run kubectl apply -f ${BATS_TESTS_DIR}/bad/bad_ns.yaml
   assert_match 'denied the request' "$output"
   assert_failure
+
+  run kubectl apply -f ${BATS_TESTS_DIR}/constraints/all_ns_must_have_gatekeeper-dryrun.yaml
+  assert_success
+
+  wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl get k8srequiredlabels.constraints.gatekeeper.sh ns-must-have-gk -o json | jq '.spec.enforcementAction' | grep dryrun"
+
+  # deploying a violation with dryrun enforcement action will be accepted
+  run kubectl apply -f ${BATS_TESTS_DIR}/bad/bad_ns.yaml
```

Comment:
we may want to use a `wait_for_process` against the `kubectl apply` here to avoid a race condition, as checking for dryrun in the spec just waits for the API server to have stored the user intent (not for the controller to have processed it). `kubectl apply` should be sufficient for that, as storing user intent is a blocking operation IIUC.

Another option would be to wait for observedGeneration to match the "generation" field in metadata. That might be a good general-purpose library function, as that basically answers the question "has Gatekeeper seen and processed the most recent version of the constraint?"

Reply:
added comparison for observedGeneration and generation